src/components/Site/Logger.tsx

```diff
@@ -32,21 +36,58 @@ export function LoggerTab({
     logger,
     logger_uid,
 }: LoggerTabProps): ReactElement {
+    const channelTemplates = useSelector((state: RootState) => state.templates);
+
     const [infoExpanded, setInfoExpanded] = useState(false);
 
     const handleInfoButton = () => setInfoExpanded(!infoExpanded);
 
-    const columns: GridColDef[] = [
-        { field: 'timestamp', headerName: 'timestamp', flex: 1 },
+    const dateFormat = 'M/D/YYYY hh:mm:ss:SSS A';
+
+    const [
+        gridRef,
+        setGridRef,
+    ] = useState<MutableRefObject<TypeComputedProps | null> | null>(null);
+
+    const columns: any[] = [
+        {
+            name: 'timestamp',
+            header: 'Timestamp',
+            minWidth: 150,
+            defaultFlex: 1,
+            dateFormat: dateFormat,
+            filterEditor: DateFilter,
+            filterEditorProps: () => {
+                return { dateFormat: dateFormat };
+            },
+        },
     ];
 
+    const filters = [
+        { name: 'timestamp', operator: 'after', type: 'date', value: '' },
+    ];
+
+    for (const [id, data] of Object.entries(channelTemplates)) {
+        const template = data as ChannelTemplate;
+
+        //populate columns based on first entry
+        if (logger.channelTemplate == id) {
+            template.keys.forEach((key) => {
+                if (key != 'timestamp') {
+                    columns.push({ name: key, header: key, defaultFlex: 1 });
+                }
+            });
+        }
+    }
+
     let rows: any[] = [];
 
     logger.data.forEach((dataPoint, index) => {
-        rows.push({
-            id: index,
-            timestamp: dataPoint.timestamp,
-        });
+        var row = Object.assign({}, dataPoint);
+        row['id'] = index;
+        row['timestamp'] = moment(dataPoint.timestamp).format(dateFormat);
+
```

Comment:
Boolean fields do not display properly, they might require an explicit toString call.

Reply:
Added a type check using the definitions from the channel templates. Couldn't use toString because the object was unknown type.  JSON.stringify() worked nicely though.